src/Java.Interop.Tools.TypeNameMappings/Java.Interop.Tools.TypeNameMappings/JavaNativeTypeManager.cs

```diff
@@ -753,6 +753,14 @@ internal static bool IsNonStaticInnerClass (TypeDefinition? type, IMetadataResol
 
 			return false;
 		}
+
+		static bool HasTypeRegistrationAttribute (TypeDefinition type)
+		{
+			if (!type.HasCustomAttributes)
+				return false;
+			return type.AnyCustomAttributes (typeof (RegisterAttribute)) ||
+				type.AnyCustomAttributes ("Java.Interop.JniTypeSignatureAttribute");
```

Comment:
Is this going to cause any issues for `XAJavaInterop1`? Do we need to find a way to ensure that this is only used for `JavaInterop1`?

Reply:
> Is this going to cause any issues for `XAJavaInterop1`?

No.  `JavaNativeTypeManager` allows obtaining the Java Callable Wrapper type name from a `System.Type` or `Mono.Cecil.TypeDefinition` instance.  `jcw-gen` already uses `[JniTypeSignature]`, as does `JavaNativeTypeManager`:

https://github.com/dotnet/java-interop/blob/87fdb8e6e385807b48b3d6aa48fdebcb26462c27/src/Java.Interop.Tools.TypeNameMappings/Java.Interop.Tools.TypeNameMappings/JavaNativeTypeManager.cs#L566-L568

The change *here* is for `IsNonStaticInnerClass()`, which didn't yet check for `[JniTypeSignature]`.

> Do we need to find a way to ensure that this is only used for `JavaInterop1`?

No.